### PR TITLE
feat(wallet): Add currency selector for buy tab

### PR DESCRIPTION
**Pull request by muliswilliam**
This PR adds functionality to allow users to select their preferred currency in the buy tab. The selected currency is used in constructing onramp provider's checkout url.

**Note**: Ramp.Network uses a users location(Country) to set currency in their checkout widget. If the country's currency is different from `fiatCurrency` query parameter in the URL, user's specified amount will be converted to country's currency.

Resolves https://github.com/brave/brave-browser/issues/23295

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

https://user-images.githubusercontent.com/48117473/174025925-baeb35b8-d334-4d45-8802-c7b8f00c1b52.mov

### CODE REVIEW

File: components/brave_wallet_ui/components/buy-send-swap/select-currency/select-currency.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    }
+  }, [onSelectCurrency])
+
+  // effects
+  React.useEffect(() => {
+    if (filteredCurrencies.length === 0 && currencies.length > 0) {
+      setFilteredCurrencies(currencies)
+      dispatch(WalletActions.selectCurrency(currencies[0]))
+    }
+  }, [currencies])
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/buy-send-swap/select-currency/select-currency.tsx:78]: Missing `filteredCurrencies` dependency

**Review by Douglashdaniel — CHANGES_REQUESTED** [review]

**Review by Douglashdaniel — APPROVED** [review]
++ Front-end

File: components/brave_wallet/browser/blockchain_registry.cc [hunk 2]
```diff
@@ -131,25 +131,61 @@ void BlockchainRegistry::GetBuyTokens(mojom::OnRampProvider provider,
...
+    std::move(callback).Run(url, "UNSUPPORTED_ONRAMP_PROVIDER");
+  }
+}
+
+void BlockchainRegistry::GetBuyUrlV1(mojom::OnRampProvider provider,
+                                     const std::string& chain_id,
+                                     const std::string& address,
+                                     const std::string& symbol,
+                                     const std::string& amount,
+                                     const std::string& currencyCode,
```
[the remark this answers is not in the document]
Comment (muliswilliam) [reply components/brave_wallet/browser/blockchain_registry.cc]: @onyb Fixed here [here]([use snake_case](https://github.com/brave/brave-core/pull/13811/commits/645079295c02244516c62fa9f3b7e11ceec75e4a))

File: components/brave_wallet/browser/blockchain_registry.cc [hunk 3]
```diff
@@ -131,25 +131,61 @@ void BlockchainRegistry::GetBuyTokens(mojom::OnRampProvider provider,
...
+// TODO(muliswilliam) - Remove this function when iOS and Android no longer
+// depend on it https://github.com/brave/brave-browser/issues/23503
 void BlockchainRegistry::GetBuyUrl(mojom::OnRampProvider provider,
                                    const std::string& chain_id,
                                    const std::string& address,
                                    const std::string& symbol,
                                    const std::string& amount,
                                    GetBuyUrlCallback callback) {
   std::string url;
+  const std::string defaultCurrency = "USD";
```
[the remark this answers is not in the document]
Comment (muliswilliam) [reply components/brave_wallet/browser/blockchain_registry.cc]: @onyb Fixed here [here](https://github.com/brave/brave-core/pull/13811/commits/645079295c02244516c62fa9f3b7e11ceec75e4a)

**Review by onyb — APPROVED** [review]
LGTM

**Pull request merged**